tests/test_run_reference_vmm.py

```diff
@@ -66,34 +82,48 @@ def test_reference_vmm(kernel):
     except subprocess.TimeoutExpired:
         # The process is still alive.
         pass
+
     assert(process_exists(vmm_process.pid))
 
-    # Poll process for new output until we find the hello world message.
-    # If we do not find the expected message, this loop will not break and the
-    # test will fail when the timeout expires.
-    expected_string = "Hello, world, from the rust-vmm reference VMM!"
-    while True:
-        nextline = vmm_process.stdout.readline()
-        if expected_string in nextline.decode():
-            break
+    return vmm_process
+
 
+def shutdown(vmm_process):
     vmm_process.stdin.write(b'reboot -f\n')
     vmm_process.stdin.flush()
 
     # If the process hasn't ended within 3 seconds, this will raise a
     # TimeoutExpired exception and fail the test.
     vmm_process.wait(timeout=3)
 
+def expect_string(vmm_process, expected_string):
+    while True:
+        nextline = vmm_process.stdout.readline()
+        if expected_string in nextline.decode():
+            break
+
+
+@pytest.mark.parametrize("kernel", KERNELS_INITRAMFS)
+def test_reference_vmm(kernel):
+    """Start the reference VMM and verify that it works."""
+
+    vmm_process = start_vmm_process(kernel, None)
+
+    # Poll process for new output until we find the hello world message.
+    # If we do not find the expected message, this loop will not break and the
+    # test will fail when the timeout expires.
+    expected_string = "Hello, world, from the rust-vmm reference VMM!"
+    expect_string(vmm_process, expected_string)
+
+    shutdown(vmm_process)
+
 
 @pytest.mark.parametrize("kernel,disk", KERNELS_DISK)
 def test_reference_vmm_with_disk(kernel, disk):
-    """Start the reference VMM and verify that it works."""
+    """Start the reference VMM with a block device and verify that it works."""
 
-    # TODO: that's actually a lie, this test is a placeholder that just
-    # verifies that the prerequisites are in place.
+    vmm_process = start_vmm_process(kernel, disk)
 
-    kernel_path = resource_path("kernel", kernel)
-    disk_path = resource_path("disk", disk)
+    expect_string(vmm_process, "Ubuntu 20.04 LTS ubuntu-rust-vmm ttyS0")
```

Comment:
Should we also run the `lsblk --json` and check fields such as:
```"ro":false && "name":"vda"```

Reply:
Would be great if we can merge the current iteration of the test, and improve it later (created #47).